src/interfaces/IProtocolFees.sol

```diff
@@ -2,14 +2,13 @@
 pragma solidity ^0.8.19;
 
 import {Currency} from "../types/Currency.sol";
+import {IProtocolFeeController} from "./IProtocolFeeController.sol";
 
 interface IProtocolFees {
     /// @notice Thrown when not enough gas is provided to look up the protocol fee
     error ProtocolFeeCannotBeFetched();
     /// @notice Thrown when the call to fetch the protocol fee reverts or returns invalid data.
     error ProtocolFeeControllerCallFailedOrInvalidResult();
```

Comment:
This only gets thrown in PoolManager. Wasn't sure if we should move it to the IPoolManager interface or keep it in IProtocolFees since it's related to protocol fees

Reply:
good question... i guess in IPoolManager because its not thrown in ProtocolFees

Reply:
Really what we need to do is move any protocol fee related code into ProtocolFees so then it can be in IProtocolFees. Im happy to have this be resolved in a separate issue though

Reply:
I was looking into it and its tough with our structure right now since `setProtocolFee` in PoolManager.sol calls `setProtocolFee` on the pool. So if we moved that function into ProtocolFees, ProtocolFees would need to have access to the pools mapping

Reply:
yeah +1 to diana she showed me what that PR would look like and it was pretty horrified to have to move the pools mapping into the protocol fees contract 😢

Reply:
Note that we chatted offline to try this in a sep PR.